doc/40-Configuration-and-Resources.md

```diff
@@ -0,0 +1,267 @@
+        $form = new DatabaseConfigForm();
+        $form->populate($moduleConfig->getSection('database'))
+            ->on(DatabaseConfigForm::ON_SUCCESS, function ($form) use ($moduleConfig) {
+                $moduleConfig->setSection('database', $form->getValues());
+                $moduleConfig->saveIni();
+
+                Notification::success('New configuration has successfully been stored');
+            })->handleRequest($this->getServerRequest());
+
+        $this->mergeTabs($this->Module()->getConfigTabs()->activate('database'));
+
+        $this->addContent($form);
+    }
+
+    /**
+     * So that we see the Module Tab and the config tab together
+     */
+    protected function mergeTabs(Tabs $tabs): void
+    {
+        foreach ($tabs->getTabs() as $tab) {
+            $this->tabs->add($tab->getName(), $tab);
+        }
+    }
+}
+```
+
+This will load the current values from from the `database` section and populate the DataBaseConfigForm with it. On saving it will store the given values in the `database` section.
+
+Everything will get written to `/etc/icingaweb2/modules/training/config.ini`
+
+We can now open our DatabaseConfigForm we can enter `assets.sqlite` and save. Our config should now look like this:
+
+```
+[Database]
+dbfile = "assets.sqlite"
+```
+
+Anywhere in our module we can now use the `Icinga\Application\Config` class to retrieve this data.
+
+```php
+use Icinga\Application\Config;
+
+// get('section', 'entry')
+Config::module('training')->get('database', 'foobar');
+Config::module('anothermodule')->get('db', 'resource');
+```
+
+## Database Refactor
+
+We can use this new config to refactor our database connection to be in a common place:
+
+In `library/Training/Database.php` we can create a common database setup:
+
+```php
+<?php
+
+namespace Icinga\Module\Training;
+
+use Icinga\Application\Config;
+use Icinga\Application\Icinga;
+use ipl\Sql\Connection;
+
+final class Database
+{
+    private static $db;
+
+    public static function get()
+    {
+        $filename = Config::module('training')->get('database', 'dbfile');
+        $dir = Icinga::app()->getModuleManager()->getModule('training')->getConfigDir();
+
+        if (self::$db === null) {
+            self::$db = new Connection([
+                'db' => 'sqlite',
+                'dbname' => $dir . '/'. $filename
+            ]);
+        }
+
+        return self::$db;
+    }
+}
+```
+
+Now we can simply use this to get our database connection:
+
+```php
+use Icinga\Module\Training\Database;
+
+$db = Database::get()
+```
+
+And we can replace the `getDb()` method in our controllers with this:
+
+```php
+public function init()
+{
+    $this->db = Database::get();
+}
+```
+
+# Resources
+
+Instead of having data sources such as databases scattered across many modules, Icinga Web provides a central `resources.ini` file that contains information about data sources. These can be referenced in other configuration files.
```

Comment:
```suggestion
Instead of having data sources - such as databases - scattered across many modules, Icinga Web provides a central `resources.ini` file that contains information about data sources. These can be referenced in other configuration files.
```